draft/planner.go

```diff
@@ -8,13 +10,23 @@ import (
 	"github.com/paketo-buildpacks/packit"
 )
 
+// A Planner sorts buildpack plan entries using a given list of priorities. A
+// Planner can also give the OR merged state of launch and build fields that
+// are defined in the buildpack plan entries metadata field.
 type Planner struct {
 }
 
+// NewPlanner returns a new Planner object.
 func NewPlanner() Planner {
 	return Planner{}
 }
 
+// Resolve takes the name of buildpack plan entries that you want to sort, the
+// buildpack plan entries that you want to be sorted, and a priority list of
+// version-sources where the 0th index is the highest priority. It returns the
+// highest priority entry as well as the sorted and filtered list of buildpack
+// plan entries that were given. Entries with no given version-source are the
+// lowest priority.
```

Comment:
It is maybe worth explaining that the priorities can either be exact string matches to the version-source or regular expressions.